src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -73,7 +73,11 @@ sealed class NumericRange[T](
 
   // See comment in Range for why this must be lazy.
   override lazy val length: Int = NumericRange.count(start, end, step, isInclusive)
-  override def isEmpty = length == 0
+  override def isEmpty = (
+    (start > end && num.gt(step, num.zero))
+      || (start < end && num.lt(step, num.zero))
+      || (start == end && !isInclusive)
+    )
   override def last: T =
     if (length == 0) Nil.head
```

Comment:
Should this now test `isEmpty`?

Reply:
@som-snytt , I've updated to replace all instances of `length ==0` with isEmpty. As far as I can tell, they're all appropriate replacements. But let me know if they're not.